### PR TITLE
AVX128: Move moves!

### PR DESCRIPTION
**Pull request by Sonicadvance1**
Also the Binary/Unary helper since this is the first use of it.

### CODE REVIEW

File: FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp [hunk 1]
```diff
@@ -617,4 +652,134 @@ void OpDispatchBuilder::AVX128_VZERO(OpcodeArgs) {
...
+
+  RefPair Src {};
+  if (IsSrcGPR) {
+    Src = AVX128_LoadSource_WithOpSize(Op, Op->Src[0], Op->Flags, !Is128Bit);
+  } else {
+    // Accesses from memory are a little weird.
+    // 128-bit operation only loads 8-bytes.
+    // 256-bit operation loads a full 32-bytes.
+    if (Is128Bit) {
+      Src.Low = LoadSource_WithOpSize(FPRClass, Op, Op->Src[0], OpSize::i128Bit, Op->Flags);
```
Comment (alyssarosenzweig) [inline FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp:727]: Isn't this loading 16 bytes? The comment says 8.

File: FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp [hunk 2]
```diff
@@ -617,4 +652,134 @@ void OpDispatchBuilder::AVX128_VZERO(OpcodeArgs) {
...
+    } else {
+      Src = AVX128_LoadSource_WithOpSize(Op, Op->Src[0], Op->Flags, !Is128Bit);
```
Comment (alyssarosenzweig) [inline FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp:729]: `!Is128Bit` is a funny way of writing `false`.

File: FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp [hunk 3]
```diff
@@ -617,4 +652,134 @@ void OpDispatchBuilder::AVX128_VZERO(OpcodeArgs) {
...
+
+  if (Op->Src[0].IsGPR()) {
+    Src = AVX128_LoadSource_WithOpSize(Op, Op->Src[0], Op->Flags, false);
+    if (ElementSize != OpSize::i128Bit) {
+      // Only duplicate if not VBROADCASTF128.
+      Src.Low = _VDupElement(OpSize::i128Bit, ElementSize, Src.Low, 0);
+    }
+  } else {
+    // Get the address to broadcast from into a GPR.
+    Ref Address = MakeSegmentAddress(Op, Op->Src[0], CTX->GetGPRSize());
```
Comment (alyssarosenzweig) [inline FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp:772]: How can 64-bit AVX instructions have segment registers?

Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp:772]: FS and GS are still a thing for memory accesses.

Comment (alyssarosenzweig) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp:772]: Oh. TLS. Kewl.

File: FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp (hunk 1 above)
Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp:727]: oop, probably slipped through when I was testing behaviour. Will fix.

Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp]: Fixed

File: FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp (hunk 2 above)
Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher/AVX_128.cpp]: Fixed

**Review by alyssarosenzweig — APPROVED** [review]

**Pull request merged**